### PR TITLE
Add the option to export simulator device logs

### PR DESCRIPTION
**Pull request by lyndsey-ferguson**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description
This adds support for the `snapshot` action to output simulator device logs.

### Motivation

Sometimes it is helpful if there are problems with the screenshots to debug what went wrong in the application. Copying the system logs for the simulated device will provide the logging that the application outputs and can help narrow down such problems.

## Testing

1. I created a unit test
2. I created a Fastfile in the example folder with a lane `:snappy` and ran it locally. The screenshots and the log files were created.

### CODE REVIEW

**Comment by lyndsey-ferguson** [comment]
This is not yet ready

**Comment by lyndsey-ferguson** [comment]
@hjanuschka can you take a look, and if all is good, merge this change?

File: fastlane/spec/fixtures/plist/Info.plist [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
```
Comment (hjanuschka) [inline fastlane/spec/fixtures/plist/Info.plist]: is that by accident or does newer xcode generate it without Computer?

File: snapshot/spec/runner_spec.rb [hunk 2]
```diff
@@ -7,5 +7,27 @@
         expect(helper_version).to match(/^SnapshotHelperVersion \[\d.\d\]$/)
       end
     end
+
+    describe 'output_simulator_logs' do
+      before(:each) do
+        allow(FileUtils).to receive(:cp).with(anything, anything) do
+          # nothing
+        end
```
Comment (hjanuschka) [inline snapshot/spec/runner_spec.rb]: please remove the block.
`allow(FileUtils).to receive(:cp).with(anything, anything)` should do it.

**Review by hjanuschka — CHANGES_REQUESTED** [review]
thx for your contribution, looking good to me, just a couple of comments left.

File: fastlane/spec/fixtures/plist/Info.plist (hunk 1 above)
Comment (lyndsey-ferguson) [inline fastlane/spec/fixtures/plist/Info.plist]: Must have come from Xcode 8.1

**Comment by lyndsey-ferguson** [comment]
@hjanuschka comments addressed. Ready to go! 🚀

File: fastlane/spec/fixtures/plist/Info.plist (hunk 1 above)
Comment (hjanuschka) [inline fastlane/spec/fixtures/plist/Info.plist]: thx - i think if it is not required for the PR it should be in original state. thx for reverting.

**Review by hjanuschka — APPROVED** [review]

**Comment by hjanuschka** [comment]
many thx for this PR!

**Comment by russbishop** [comment]
I would suggest `xcrun simctl spawn booted log collect`. The output will be written to `$(xcrun simctl getenv booted SIMULATOR_SHARED_RESOURCES_DIR)/system_logs.logarchive`

As more and more things move to `os_log` the old text log file may become less useful. Some debugging information is already stored in an in-memory ring buffer.

**Comment by hjanuschka** [comment]
@russbishop that sounds awesome, i am ready to go for it. 

just to make sure i got it:
  * launch snapshot, doe tests on simulator
  * collect screenshots
  * run `xcrun simctl spawn booted log collect`
  * get log folder via `xcrun simctl getenv booted SIMULATOR_SHARED_RESOURCES_DIR`
  * copy over `path/system_logs.logarchive`

does this include the "normal" NSLog, or do we need to fetch the `system.log` too to get them?

would love to hear back on this.

**Comment by hjanuschka** [comment]
@russbishop  carried on in https://github.com/fastlane/fastlane/pull/8033 maybe you can have a look at the code.

**Comment by russbishop** [comment]
@hjanuschka Yes, that's correct. The new system captures Syslog, Apple System Logger, and the new `os_log` messages all in one place. I would expect everything to switch to the new logging system over time.

I updated the env var; I should have spelled the word "directory" out fully.

**Pull request merged**